Apps/WebClient/src/ClientApp/app/views/timeline.vue

```diff
@@ -220,12 +247,44 @@ export default class TimelineComponent extends Vue {
     return moment(date).format("ll");
   }
 
+  @Watch("filterText")
+  private filter() {
+    if (!this.filterText) {
+      this.visibleTimelineEntries = this.timelineEntries;
+      return;
+    }
+    if (this.filterTimeout) {
+      clearTimeout(this.filterTimeout);
+    }
+
+    this.filterTimeout = setTimeout(() => {
+      console.log(`filtering ${this.filterText}`);
+      this.visibleTimelineEntries = this.timelineEntries.filter(
+        this.filterExpression
+      );
+    }, 1000);
+  }
+
+  private filterExpression(entry: TimelineEntry) {

```

Comment:
Since this will need to be done for each type of component, lets move it to the medicationentry component.